### PR TITLE
Incorporate did_web test vectors

### PR DESCRIPTION
**Pull request by andresuribe87**
# Overview
This PR adds the test vectors used in sdk-development for did-web, and updates code so that the tests pass.

# How Has This Been Tested?
Unit tests following the SDK conventions.

### CODE REVIEW

File: credentials/src/test/kotlin/web5/sdk/credentials/PresentationExchangeTest.kt [hunk 1]
```diff
@@ -450,8 +450,8 @@ class PresentationExchangeTest {
 
       val selectedCreds = PresentationExchange.selectCredentials(listOf(vcJwt), pd)
 
-      assertEquals( 1, selectedCreds.size)
```
Comment (nitro-neal) [inline credentials/src/test/kotlin/web5/sdk/credentials/PresentationExchangeTest.kt:453]: err I wonder why my formatter is doing this.

for web5-js we have a prettier task - https://github.com/TBD54566975/web5-js/issues/266

should do something similar across all our repos so formatting stays in sync

File: dids/build.gradle.kts [hunk 2]
```diff
@@ -38,4 +38,5 @@ dependencies {
 
   implementation("dnsjava:dnsjava:3.5.2")
   implementation("com.github.mpetazzoni:ttorrent:ttorrent-2.0")
+  testImplementation(project(mapOf("path" to ":testing")))
```
Comment (nitro-neal) [inline dids/build.gradle.kts:41]: why was this needed?

File: dids/src/main/kotlin/web5/sdk/dids/DidResolutionResult.kt [hunk 3]
```diff
@@ -18,20 +18,48 @@ import web5.sdk.dids.methods.ion.models.MetadataMethod
...
 
-  private companion object {
+  override fun equals(other: Any?): Boolean {
+    if (other is DidResolutionResult) {
+      return this.toString() == other.toString()
+    }
+    return false
+  }
+
+  override fun hashCode(): Int {
```
Comment (nitro-neal) [inline dids/src/main/kotlin/web5/sdk/dids/DidResolutionResult.kt:37]: chad says we can do this:

```
override fun hashCode(): Int {
    return hash(context, didDocument, didDocumentMetadata, didResolutionMetadata)
}

```

but idk if that actually works

File: dids/src/main/kotlin/web5/sdk/dids/methods/web/DidWeb.kt [hunk 4]
```diff
@@ -117,12 +120,25 @@ public sealed class DidWebApi(
   override val methodName: String = "web"
 
   override fun resolve(did: String, options: ResolveDidOptions?): DidResolutionResult {
-    val docURL = getDocURL(did)
+    val parsedDid = try {
+      DID.fromString(did)
+    } catch (_: ParserException) {
```
Comment (nitro-neal) [inline dids/src/main/kotlin/web5/sdk/dids/methods/web/DidWeb.kt:125]: should we have another line to catch all other exceptions?
```
} catch (e: Exception) {
```

**Review by nitro-neal — APPROVED** [review]
🍰

**Comment by andresuribe87** [comment]
@phoebe-lew @mistermoe or @tomdaffurn would appreciate a review here. Thanks!

File: test-vectors/did_web/resolve.json [hunk 5]
```diff
@@ -70,6 +75,8 @@
         "didUri": "did:web:doesnotexist.com"
```
Comment (frankhinek) [inline test-vectors/did_web/resolve.json:75]: What are your thoughts on returning `internalError` instead of `notFound` when the HTTP GET fails?

Seems like that's the direction based on reading this: https://github.com/TBD54566975/web5-kt/pull/172/files#diff-51da16f10d7795c05d587a710d194142c4addebe8657e5abd9570ac6fc4cc738R64-R73

Comment (andresuribe87) [reply test-vectors/did_web/resolve.json:75]: @frankhinek did you mean to link to a commit done within this PR?

My thoughts are that notFound carries a different semantic meaning than internal error. Programmatically, I'd expect retries to be done with internal errors. I wouldn't expect that to happen with not found. 

Given that, I'd like to keep as is. I'm open to hearing other opinions. Can you share your rationale?

File: credentials/src/test/kotlin/web5/sdk/credentials/PresentationExchangeTest.kt (hunk 1 above)
Comment (phoebe-lew) [reply credentials/src/test/kotlin/web5/sdk/credentials/PresentationExchangeTest.kt:455]: weird, thought this would have been taken care of by ktlint/detekt

File: dids/src/main/kotlin/web5/sdk/dids/exceptions/PkarrRecordResponseException.kt [hunk 6]
```diff
@@ -6,3 +6,10 @@ package web5.sdk.dids.exceptions
  * @param message the exception message detailing the error
  */
 public class PkarrRecordResponseException(message: String) : RuntimeException(message)
+
+/**
+ * Did resolution exception.
+ *
+ * @param message the exception message detailing the error
+ */
+public class DidResolutionException(message: String) : RuntimeException(message)
```
Comment (phoebe-lew) [inline dids/src/main/kotlin/web5/sdk/dids/exceptions/PkarrRecordResponseException.kt:15]: Should this be in its own exception file?

File: test-vectors/did_web/resolve.json
Comment (phoebe-lew) [inline test-vectors/did_web/resolve.json:1]: follow up PR to incorporate changes back into `sdk-development`?

**Review by phoebe-lew — APPROVED** [review]

File: dids/src/main/kotlin/web5/sdk/dids/exceptions/PkarrRecordResponseException.kt (hunk 6 above)
Comment (andresuribe87) [reply dids/src/main/kotlin/web5/sdk/dids/exceptions/PkarrRecordResponseException.kt:15]: Done in [cb911e9](https://github.com/TBD54566975/web5-kt/pull/172/commits/cb911e9a37ca2de4f8e063e6bb1cab7e084d8fb9)

File: test-vectors/did_web/resolve.json
Comment (andresuribe87) [reply test-vectors/did_web/resolve.json:1]: This PR took the test vectors from `sdk-development`, so no changes are needed on that side.
